flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/partitioner/StreamPartitionerTest.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.runtime.partitioner;
+
+import org.apache.flink.api.java.tuple.Tuple;
+import org.apache.flink.runtime.plugable.SerializationDelegate;
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Tests for different {@link StreamPartitioner} implementations.
+ */
+public class StreamPartitionerTest extends TestLogger {
+
+	protected StreamPartitioner<Tuple> streamPartitioner;
+	protected StreamRecord<Tuple> streamRecord = new StreamRecord<>(null);
+	protected SerializationDelegate<StreamRecord<Tuple>> sd = new SerializationDelegate<>(null);
+
+	@Test
+	public void testSelectChannelsLength() {
+		sd.setInstance(streamRecord);
```

Comment:
put it in
```
@Before
public void setUp() {
  serializationDelegate.setInstance(streamRecord);
}
```
This way you deduplicate this setup logic from every test as well.

Reply:
👍